compute_tools/src/extension_server.rs

```diff
@@ -123,16 +123,41 @@ pub async fn get_available_extensions(
         .download_stream
         .read_to_end(&mut ext_idx_buffer)
         .await?;
-    let ext_index_str = str::from_utf8(&ext_idx_buffer).expect("error parsing json");
+    let ext_index_str = str::from_utf8(&ext_idx_buffer).expect("json err");
     let ext_index_full: Value = serde_json::from_str(ext_index_str)?;
-    let ext_index_full = ext_index_full.as_object().context("error parsing json")?;
+    let ext_index_full = ext_index_full.as_object().expect("json err");
     info!("ext_index: {:?}", &ext_index_full);
 
+    let public_extensions = ext_index_full["public_extensions"]
+        .as_array()
+        .expect("json err");
+    // TODO: maybe enabled_extensions should be a HashSet? probably doesn't matter
+    let mut enabled_extensions = public_extensions
+        .iter()
+        .map(|x| x.as_str().expect("json err"))
+        .collect::<Vec<&str>>();
+    for custom_extension in custom_extensions {
+        enabled_extensions.push(custom_extension);
+    }
+
+    let library_index = ext_index_full["library_index"]
+        .as_object()
+        .context("error parsing json")?;
+    let mut parsed_lib_index = HashMap::new();
+    for (key, val) in library_index {
+        let parsed_val = val.as_str().expect("json err").to_string();
+        parsed_lib_index.insert(key.to_string(), parsed_val);
+    }
+
+    let all_extension_data = ext_index_full["extension_data"]
+        .as_object()
+        .context("error parsing json")?;
```

Comment:
Using the dom model (serde_json::Value) has no upsides, only downsides.

```rust
#[derive(Debug, serde::Deserialize)]
struct Index {
  public_extensions: Vec<String>,
  library_index: HashMap<String, String>,
  extension_data: HashMap<String, ExtensionData>,
}

#[derive(Debug, serde::Deserialize)]
struct ExtensionData {
  control_data: HashMap<String, String>,
  archive_path: String,
}
```

I think that this is what your data looks like. You can parse it as `serde_json::from_slice::<Index>(ext_idx_buffer);`.

Now it's a much easier place to document what does a thing mean for example, and you don't need to repeat the `.expect("json err")`.